jcommune-service/src/test/java/org/jtalks/jcommune/service/bb2htmlprocessors/UrlToLinkConvertPostProcessorTest.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (C) 2011  JTalks.org Team
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package org.jtalks.jcommune.service.bb2htmlprocessors;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+
+/**
+ * Created by Alexey Usharovskiy on 28.12.16.
+ */
+public class UrlToLinkConvertPostProcessorTest {
+    UrlToLinkConvertPostProcessor postProcessor = new UrlToLinkConvertPostProcessor();
+
+    @Test
+    void doesNothingIfNoLinksPresent() {
+        String originalText = "Silent sir say desire fat him letter. Whatever settling goodness too and \n" +
+                "honoured she building answered her. Strongly thoughts remember mr to do consider debating. \n" +
+                "Spirits musical behaved on we he farther letters. Repulsive he he as deficient newspaper dashwoods we. \n" +
+                "Discovered her his pianoforte insipidity entreaties. Began he at terms meant as fancy. Breakfast \n" +
+                "arranging he if furniture we described on. Astonished thoroughly unpleasant especially you dispatched \n" +
+                "bed favourable.";
+        assertEquals(postProcessor.postProcess(originalText), originalText);
+    }
+
+    @Test
+    void doesNothingIfNoTextPresent() {
+        assertEquals(postProcessor.postProcess(""), "");
+    }
+
+    @Test(dataProvider = "urlShouldBeHighlighted")
+    void doesUrlHighlightingIfUrlShouldBeHighlighted(String originalText, String expectedOutput) {
+        assertEquals(postProcessor.postProcess(originalText), expectedOutput);
+    }
+
+    @Test(dataProvider = "urlShouldNotBeHighlighted")
+    void doesNoUrlHighlightingIfUrlShouldNotBeHighlighted(String originalText, String expectedOutput) {
```

Comment:
`insertsNoHrefTags_ifThereAreNoLinks`

Also, there shouldn't be expectedOutput, right? It must always be the same as `originalText` in this case.